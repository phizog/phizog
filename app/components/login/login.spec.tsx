/* eslint-env jest */
import * as React from 'react'
import { Login } from '.'
import { mount, shallow } from 'enzyme'
import { MemoryRouter } from 'react-router'

describe('<Login />', () => {
  let mock: any = jest.fn()

  it('should render correctly', () => {
    const component = shallow(
      <MemoryRouter>
        <Login
          inProgress={true}
          toggle={mock}
          profile={mock}
          history={mock}
          location={mock}
          match={mock}
          staticContext={mock}
        />
      </MemoryRouter>
    )

    expect(component).toMatchSnapshot()
  })
})
