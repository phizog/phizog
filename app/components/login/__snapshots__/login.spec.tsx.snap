// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MemoryRouter>
    <Login
      history={[MockFunction]}
      inProgress={true}
      location={[MockFunction]}
      match={[MockFunction]}
      profile={[MockFunction]}
      staticContext={[MockFunction]}
      toggle={[MockFunction]}
    />
  </MemoryRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Login
        history={[MockFunction]}
        inProgress={true}
        location={[MockFunction]}
        match={[MockFunction]}
        profile={[MockFunction]}
        staticContext={[MockFunction]}
        toggle={[MockFunction]}
      />,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "8fh4tu",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "8fh4tu",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "history": [MockFunction],
        "inProgress": true,
        "location": [MockFunction],
        "match": [MockFunction],
        "profile": [MockFunction],
        "staticContext": [MockFunction],
        "toggle": [MockFunction],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Login
          history={[MockFunction]}
          inProgress={true}
          location={[MockFunction]}
          match={[MockFunction]}
          profile={[MockFunction]}
          staticContext={[MockFunction]}
          toggle={[MockFunction]}
        />,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "8fh4tu",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "8fh4tu",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "history": [MockFunction],
          "inProgress": true,
          "location": [MockFunction],
          "match": [MockFunction],
          "profile": [MockFunction],
          "staticContext": [MockFunction],
          "toggle": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
